docs/concurrency.org

```diff
@@ -0,0 +1,79 @@
+Note: il se peut qu je dise des trucs faux car la plupart des ces termes sont complétement nouveaux, donc si tu vois des trucs louches, cest assez probable que tu aies un bon feeling.
+
+Je dois voir Alberto lors de ma prochaine séance pour lui poser plusieurs questsions sur ces différents termes etc
+
+** Definitions
+Coroutine (cooperative multitasking): 
+- a coroutine itself is not concurrent !, it allows to manage concurrent tasks in the same way as callbacks for example
+- single threaded
+- allows to write code in a sequential manner
+- allows not to have context switch
+- 2 types of coroutine: symetric vs asymetric (relative to the control-transfer mechanism)
+- coroutines are not preemptive, they are cooperative, therefore they must explicitly give back control by **yielding**
+- usually uses keywords like **yield**, **resume**, **suspend**, ...
+- " With nonstackful coroutines, a task cannot yield within those libraries. Therefore, either those programs will block until the operation completes or they must be restructured to avoid doing I/O inside library functions."
+
+Fibers :
+- cooperative multitasking instead of preemptive multitasking
```

Comment:
What is the difference ?

Reply:
i tink that you meant between premptive and cooperative ? if thats the case i wrote small informations about both of them